Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,587 @@
+            .getCodeExtraInformation()
+            .setFunctionName(`get${property.functionName}`);
+        }
+
+        // Add all the generic setters
+        if (parameterType === 'number' || parameterType === 'string') {
+          const expressionType =
+            parameterType === 'number' ? 'expression' : 'string';
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _('Set ') + property.description,
+              _('Set ') + property.description,
+              _('Do _PARAM1__PARAM2_ to the ') +
+                property.description +
+                _(' of _PARAM0_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
+              objectName,
+              false
+            )
+            .addParameter('operator', _("Modification's sign"), '', false)
+            .addParameter(expressionType, property.paramLabel, '', false)
+            .getCodeExtraInformation()
+            .setFunctionName(`set${property.functionName}`)
+            .setManipulatedType(parameterType)
+            .setGetter(`get${property.functionName}`);
+        } else {
+          // Setter doesnt have a +- modifier (Color, alignment, etc)
+          gdObject
+            .addAction(
+              `Set${property.functionName}`,
+              _('Set ') + property.description,
+              _('Set ') + property.description,
+              _('Set ') + property.description + _(' of _PARAM0_ to _PARAM1_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
+              objectName,
+              false
+            )
+            .addParameter(
+              parameterType,
+              property.paramLabel,
+              property.options ? property.options : '',
+              false
+            )
+            .getCodeExtraInformation()
+            .setFunctionName(`set${property.functionName}`)
+            .setGetter(`get${property.functionName}`);
+        }
+
+        // Add compare Conditions via getters
+        if (parameterType === 'string' || parameterType === 'number') {
+          const propExpressionType =
+            parameterType === 'string' ? 'string' : 'expression';
+          gdObject
+            .addCondition(
+              `Is${property.functionName}`,
+              property.description,
+              _('Compare the value of ') + property.description,
+              _('The ') +
+                property.paramLabel +
+                _(' of _PARAM0_ is _PARAM1__PARAM2_'),
+              '',
+              property.iconPath,
+              property.iconPath
+            )
+            .addParameter(
+              'object',
+              objectName + _(' object'),
+              objectName,
```

Comment:
The third parameter is the object type, not the object name. In your case, it's the same, but in general this is not the case. For example, a particle emitter type could be "ParticleEmitter" while the name displayed to the user is "Particle Emitter" (note the space).

So it would be better to pass both the objectName (not translated) and objectFullName (translated with _(""))